clang/include/clang/Analysis/FlowSensitive/DataflowAnalysis.h

```diff
@@ -28,12 +28,24 @@
 #include "clang/Analysis/FlowSensitive/MatchSwitch.h"
 #include "clang/Analysis/FlowSensitive/TypeErasedDataflowAnalysis.h"
 #include "clang/Analysis/FlowSensitive/WatchedLiteralsSolver.h"
+#include "clang/Support/Compiler.h"
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/STLFunctionalExtras.h"
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/Support/Errc.h"
 #include "llvm/Support/Error.h"
 
+namespace clang {
+namespace dataflow {
+class NoopLattice;
```

Comment:
Why is this forward declaration placed in this file, when the NoopLattice is defined in its own file? Regardless, if it does belong here, please comment to explain the intent. I'm a code owner and yet I don't understand what this is doing and how/what I should do to prevent it from breaking in the case of changes. IIUC, this change is Window-specific and seems fragile, making it all the more critical to clearly document.

Reply:
Its used as type in a extern template declaration below,  The extern template declaration needs to be seen by code using any_cast with NoopLattice to allow external code to use the same address for the Any: Id for NoopLattice for shared library builds on windows , but i guess it really could go in the NoopLattice header.  I can make whatever changes you think would be best.

Reply:
Sorry -- I was asking about the whole block, not just the forward decl. Yes, please move it the NoopLattice header. Thanks!